web/src/components/LoginScreen.js

```diff
@@ -0,0 +1,204 @@
+  };
+
+  return (
+    <Container>
+      <Card>
+        <LoginFormRow>
+          <LoginHeading>
+            {mode === "login" ? "Log In" : "Register an Account"}
+          </LoginHeading>
+        </LoginFormRow>
+        <LoginFormRow>
+          <TextInput
+            type="email"
+            label="Email"
+            placeholder="your.email@example.com"
+            onChange={(e) => {
+              setError((e) => ({ ...e, email: undefined }));
+              setEmail(e.target.value);
+            }}
+            value={email}
+            state={
+              error.email
+                ? "error"
+                : validator.isEmail(email)
+                ? "valid"
+                : "none"
+            }
+            errorMessage={error.email}
+          />
+        </LoginFormRow>
+        <LoginFormRow>
+          <TextInput
+            type="password"
+            label="Password"
+            placeholder="pa55w0rd"
+            onChange={(e) => {
+              setPassword(e.target.value);
+            }}
+            value={password}
+            state={error.password ? "error" : error.password ? "valid" : "none"}
+            errorMessage={error.password}
+          />
+        </LoginFormRow>
+        {mode === "login" ? (
+          <Button variant="primary" onClick={() => handleLogin()}>
+            Log In
+          </Button>
+        ) : (
+          <Button
+            variant="primary"
+            onClick={() => handleRegistrationAndLogin()}
+          >
+            Register
+          </Button>
+        )}
+        <ToggleContainer>
+          <ToggleText>
+            {mode === "login"
+              ? "Don't have an account?"
+              : "Already have an account?"}
+          </ToggleText>
+          <ToggleLink
+            onClick={(e) => {
+              e.preventDefault();
+              toggleMode();
+            }}
+          >
+            {mode === "login" ? "Register one now." : "Log in instead."}
+          </ToggleLink>
+        </ToggleContainer>
+      </Card>
+    </Container>
+  );
+};
+
+function handleAuthenticationError(err, setError) {
+  const { status, message } = parseAuthenticationError(err);
+  const errorType = message || status;
+  switch (errorType) {
+    case "invalid username":
+      setError((prevErr) => ({ ...prevErr, email: "Invalid email address." }));
+      break;
+    case "invalid username/password":
+    case "invalid password":
+    case "401":
+      setError((err) => ({ ...err, password: "Incorrect password." }));
+      break;
+    case "name already in use":
+    case "409":
+      setError((err) => ({ ...err, email: "Email is already registered." }));
+      break;
+    case "password must be between 6 and 128 characters":
+    case "400":
+      setError((err) => ({
+        ...err,
+        password: "Password must be between 6 and 128 characters.",
+      }));
+      break;
+    default:
+      break;
+  }
+}
+
+function parseAuthenticationError(err) {
```

Comment:
It's kind of wonky that the web SDK returns a string as the error and forces us to manually parse it w/ regex rather than doing something like:
```
{
errorDescription: "Error: Request failed (POST https://stitch.mongodb.com/api/client/v2.0/app/tasktracker-huhcb/auth/providers/local-userpass/login): invalid username/password (status 401)",
errorMessage: "invalid username/password",
status: 401
}
```

not at all your fault, but might be something work requesting from the web-sdk engineers